### PR TITLE
Allow an exit action, to wait for a signal from a child (also cargo fmt)

### PR DESCRIPTION
**Pull request by vkrasnov**
Introduces a new function `exit_action`, which registers a callback to be run by the original process just before it exits. It is useful when one wants to wait for a signal from the forked process before the exit, or print a notification for example.

Also formatted by accident, but can create a PR without formatting if preferred.

### CODE REVIEW

**Comment by RandomInsano** [comment]
I’m not a maintainer, but the formatting changes make your intent hard to follow and as a good practice should never be allowed to be bundled with your PR. I’d remove it to help the project owners understand what you’re doing with a lot less noise and be a good code citizen.

FWIW I prefer your format settings these days and the existing formatting from a time in my life before I shared code.

**Comment by vkrasnov** [comment]
I agree, the format wasn't intentional. Just a side effect of my dev env. I could revert the format, but it doesn't look like this project is actively maintained anyway.
